azurelinuxagent/ga/policy/policy_engine.py

```diff
@@ -142,139 +113,162 @@ def __read_policy():
+        return parsed
+
+    @staticmethod
+    def _parse_extension(extension):
         """
-        extensions = extensions_policy.get("extensions")
-        if extensions is None:
-            return
+        Parses an individual extension. See _parse_policy() for schema.
+        """
+        extension_attribute_name = "extensionPolicies.extensions.{0}".format(extension)
+
+        _PolicyEngine._check_attributes(extension, object_name=extension_attribute_name, valid_attributes=["signatureRequired", "runtimePolicy"])
+
+        return_value = {}
+
+        signature_required = _PolicyEngine._get_boolean(extension, attribute="signatureRequired", name_prefix=extension_attribute_name, optional=True, default=None)
+        if signature_required is not None:
+            return_value["signatureRequired"] = signature_required
 
-        parsed_extensions_dict = {}
-        for extension_name, individual_policy in extensions.items():
+        # The runtimePolicy is an arbitrary object.
+        runtime_policy = extension.get("runtimePolicy")
+        if runtime_policy is not None:
+            return_value["runtimePolicy"] = runtime_policy
 
-            # We don't validate "extensions" against the schema, because the attributes (individual extension names)
-            # are dynamic and not defined in the schema. We do validate that individual_policy is a dict, and validate
-            # the schema of individual_policy.
-            individual_policy_schema = _POLICY_SCHEMA["extensionPolicies"]["extensions"]["<extensionName>"]
-            if not isinstance(individual_policy, dict):
-                raise InvalidPolicyError("invalid type {0} for attribute '{1}', please change to object."
-                                         .format(type(individual_policy).__name__, extension_name))
-            _PolicyEngine.__validate_schema(individual_policy, individual_policy_schema, extension_name)
+        return return_value
+
+    @staticmethod
+    def _check_attributes(object_, object_name, valid_attributes):
+        """
+        Check that the given object, which should be a dictionary, has only the specified attributes.
+        If any other attributes are present, raise InvalidPolicyError.
+        The object_name is used in the error message.
+        """
+        for k in object_.keys():
+            if k not in valid_attributes:
+                raise InvalidPolicyError("invalid attribute '{0}' in {1}".format(k, object_name))
 
-            extension_signature_policy = individual_policy.get("signatureRequired")
-            if extension_signature_policy is None:
-                extension_signature_policy = template["extensionPolicies"]["signatureRequired"]
+    @staticmethod
+    def _get_dictionary(object_, attribute, name_prefix="", optional=False, default=None):
+        """
+        Returns object[attribute] if it exists, verifying that it is a dictionary, else returns default.
```

Comment:
nit: can we update the comment to indicate this raises InvalidPolicyError if type verification fails